go/vt/worker/split_diff.go

```diff
@@ -429,13 +430,30 @@ func (sdw *SplitDiffWorker) diff(ctx context.Context) error {
 	sdw.wr.Logger().Infof("Running the diffs...")
 	// TODO(mberlin): Parameterize the hard coded value 8.
 	sem := sync2.NewSemaphore(8, 0)
-	for _, tableDefinition := range sdw.destinationSchemaDefinition.TableDefinitions {
+	tableDefinitions := sdw.destinationSchemaDefinition.TableDefinitions
+
+	// sort tables by size
+	// if there are large deltas between table sizes then it's more efficient to start working on the large tables first
+	sort.Slice(tableDefinitions, func(i, j int) bool { return tableDefinitions[i].DataLength > tableDefinitions[j].DataLength })
+
+	// use a channel to make sure tables are diffed in order
+	tableChan := make(chan *tabletmanagerdatapb.TableDefinition, len(tableDefinitions))
+	for _, tableDefinition := range tableDefinitions {
+		tableChan <- tableDefinition
+	}
+
+	// start as many goroutines as there are tables to diff
+	for range tableDefinitions {
 		wg.Add(1)
-		go func(tableDefinition *tabletmanagerdatapb.TableDefinition) {
+		go func(tableChan chan *tabletmanagerdatapb.TableDefinition) {
```

Comment:
You don't need to pass the channel as a parameter here, just use it directly.

(you only need to pass the parameters when they are the 'for' variables, because these variables are actually references, and they change while the loop is running, so you don't want a go routine to use them, as they'd use the ever-changing value)

Reply:
gotcha! done!